### ISSUE TITLE
LIM: Respect Menu Font Settings

### ISSUE THREAD

**Issue by QORTEC** [Zren/material-decoration#31]
**Current behaviour**:
Currently LIM uses the ``Window title`` font settings 

**Expected behaviour**:
LIM should use the ``Menu`` font settings

**Steps to Reproduce**:
Open KDE “System Settings”
Under ``Appearance`` click on ``Fonts``
Change the font settings for ``Menu`` and ``Window title``
Click ``Apply``

![Screenshot_20210205_223433](https://user-images.githubusercontent.com/4239654/107108248-93936380-6804-11eb-912b-f3c7aeb863a6.png)

**Comment by ben2talk** [comment]
The actual menu follows menu font. Menu items in the titlebar use titlebar font - actually this looks good to me.

**Comment by QORTEC** [comment]
@ben2talk thanks for pointing that out, don't know how I missed that.
I mainly opened the bug report since the menu was not following the expected behaviour (to me anyways).

It could be argued that:
 - since the LIM is integrated with titlebar it should; all elements in inside titlebar itself should use the ``Window title`` settings
 - since changing the ``Menu`` settings effects the entire menu in KDE by default; LIM should respect that

If the current behaviour is preferred/expected; I would like to know how feasible would it be to add a toggle between the two settings (``Window title`` & ``Menu``)?

**Comment by Zren** [comment]
I won't be adding an option, but you can modify it to do so.


* Here's where material draws the menu in the titlebar:
  * https://github.com/Zren/material-decoration/blob/master/src/TextButton.cc#L59
* Here's how the KDecoration API calculates the titlebar `QFont`:
  * https://github.com/KDE/kdecoration/blob/master/src/private/decorationsettingsprivate.cpp#L46-L49
  * https://doc.qt.io/qt-5/qfontdatabase.html#SystemFont-enum
* No idea how to get the font for the menu.
  * https://github.com/KDE/breeze/blob/master/kstyle/breezestyle.cpp#L4796
  * https://doc.qt.io/qt-5/qfont.html

It's probably easier to hardcode a `QFont` with the desired size + family.

**Comment by hexclover** [comment]
Hello, just to mention that I've implemented the option in my fork: https://github.com/hexclover/material-decoration. Feel free to test it out.

<img height=300px src="https://user-images.githubusercontent.com/47456195/181826098-31ab6dea-5aee-4971-83f5-cadc73e63d4c.png" />

With the option ON:

![image](https://user-images.githubusercontent.com/47456195/181826206-c31e84a6-23ad-4e40-b1f6-6c4ab8d40dc5.png)

With the option OFF:

![image](https://user-images.githubusercontent.com/47456195/181826412-84355887-8772-4fb3-9909-c93d4b274698.png)

**Comment by QORTEC** [comment]
@hexclover I just built and tested your Material-Decorations fork.
After some quick testing it I noticed that the `Use menu font when drawing menu items` option uses the `general` font not the `menu` font. (I use the same font for both so I'm happy with it as is.)

**System Information**:
|  |  |
| ------------- | ------------- |
| Operating System: | Solus 4.3 |
| Kernel Version: | 5.15.50-216-zfs |
| KDE Plasma Version: | 5.25.3 |
| KDE Frameworks Version: | 5.96.0 |
| QT Version: | 5.15.5 |
| Graphics Platform: | X11 |

**Comment by hexclover** [comment]
@QORTEC Hello, what was your testing procedure? Can you try the following steps:

1. In system settings, set General and Menu fonts to different fonts.
2. Logout and login again. Or, at least restart KWin.
3. Try toggling the option. Did it work?

**Comment by QORTEC** [comment]
@hexclover, it works! Kinda?
There seems to be a bug, when you change a font the _menu_ uses the `general` font.
Restarting kwin resets the _menu_ to use the `menu` font. 

**Steps to Reproduce**: 
1. change any font whatsoever 
2. apply said change

**Result**:
The _menu_ uses the `general` font.

**Expected Behaviour**:
The _menu_ uses the `menu` font.

**Workaround**:
Restart `kwin`, the _menu_  will use the `menu` font.

**Comment by hexclover** [comment]
@QORTEC Yes, there seems to be a long-standing bug in KDE (?): If you change any (?) of the font settings and apply them, applications will start to use the *General* font for *menus* until next login. I don't think this is related/specific to ```material-decoration```.
